### DISCUSSION TITLE
Connect to postgres through libpq environment variables?

### DISCUSSION THREAD

**Discussion by ianliu** [sqlalchemy/sqlalchemy#7828] [Usage Questions] [answered]
When using psycopg, I can just call `psycopg.connect()` without arguments and pass whatever credentials I want through environment variables. For example, I can set `PGSERVICE=mydb` and call my python script. Is it possible to create a SQLAlchemy engine in the same manner?

**Comment by zzzeek** [comment, marked as answer]
yes it is, see the short section at https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#empty-dsn-connections-environment-variable-connections for a short description

**Reply by ianliu** [reply]
Thanks!
